lib/request.js

```diff
@@ -138,6 +140,171 @@ function buildRequestWithTrustProxy (R, trustProxy) {
   return _Request
 }
 
+function isValidHostname (hostname) {
+  if (!hostname) {
+    return false
+  }
+  if (net.isIP(hostname)) {
+    return true
+  }
+  return /^(?=.{1,253}$)(?!-)[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)*(?<!-)$/.test(hostname)
+}
+
+function parsePortPart (strippedHost) {
+  /* No port part */
+  if (!strippedHost) {
+    return null
+  }
+
+  /* Ensure everything is a digit (includes preventing multiple port separators) */
+  if (!/^\d+$/.test(strippedHost)) {
```

Comment:
Why?

Reply:
It checks if after the port separator everything is valid (a number). For example if I define `example.com:abc` it will stop it.